src/rustllvm/RustWrapper.cpp

```diff
@@ -1000,6 +1000,20 @@ extern "C" bool LLVMRustAddYkBlockLabel(LLVMBuilderRef Builder,
     return true;
 }
 
+extern "C" unsigned int LLVMRustInstructionIndex(LLVMValueRef Value) {
+    Instruction *Instr = unwrap<Instruction>(Value);
+    BasicBlock *B = Instr->getParent();
+    unsigned int index = 0;
+    for (BasicBlock::iterator I = B->begin(), IE = B->end(); I != IE; ++I) {
+        Instruction *CI = cast<Instruction>(I);
```

Comment:
I think this can be a `static_cast`, which is more efficient, as it eliminates a type check.

https://stackoverflow.com/questions/28002/regular-cast-vs-static-cast-vs-dynamic-cast

Reply:
Okay, so I just replace `cast` with `static_cast`. Makes me wonder why the llvm code doesn't use `static_cast`s then.

Reply:
That's a good question.

Reply:
Let's leave it for now. We can always optimise later. I reckon this whole code is gonna change anyways at some point.

Reply:
I agree.

All of the docs I've read don't even mention the cast operators used in this file.

I wonder if `cast` and `dyn_cast` are `static_cast` and `dynamic_cast` renamed...

Reply:
That's certainly possible.